src/Cli/ConfigGenerator.cs

```diff
@@ -1095,6 +1115,95 @@ private static bool TryUpdateConfiguredHostValues(
             }
         }
 
+        /// <summary>
+        /// Attempts to update the Azure Log Analytics configuration options based on the provided values.
+        /// Validates that any user-provided parameter value is valid and updates the runtime configuration accordingly.
+        /// </summary>
+        /// <param name="options">The configuration options provided by the user.</param>
+        /// <param name="azureLogAnalyticsOptions">The Azure Log Analytics options to be updated.</param>
+        /// <returns>True if the Azure Key Vault options were successfully configured; otherwise, false.</returns>
+        private static bool TryUpdateConfiguredAzureLogAnalyticsOptions(
+            ConfigureOptions options,
+            [NotNullWhen(true)] ref AzureLogAnalyticsOptions? azureLogAnalyticsOptions)
+        {
+            try
+            {
+                AzureLogAnalyticsAuthOptions? updatedAuthOptions = azureLogAnalyticsOptions?.Auth;
+
+                // Runtime.Telemetry.AzureLogAnalytics.Enabled
+                if (options.AzureLogAnalyticsEnabled is not null)
+                {
+                    azureLogAnalyticsOptions = azureLogAnalyticsOptions! with { Enabled = options.AzureLogAnalyticsEnabled is CliBool.True ? true : false, UserProvidedEnabled = true };
```

Comment:
How are we certain that `azureLogAnalyticsOptions` will not be null here?

If we are certain that it can never be null, the argument to this function should be defined as non-nullable.